hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -134,14 +141,20 @@ private void handlePlatformTransaction(
 
         try {
             // Setup configuration
-            var configuration = configProvider.getConfiguration();
+            final var configuration = configProvider.getConfiguration();
             final var hederaConfig = configuration.getConfigData(HederaConfig.class);
 
             final var preHandleResult = getCurrentPreHandleResult(state, platformEvent, platformTxn, configuration);
             recordBuilder.transaction(
                     preHandleResult.txInfo().transaction(),
                     preHandleResult.txInfo().signedBytes());
 
+            // Check transaction duplication
+            checkDuplicates(preHandleResult.txInfo().txBody().transactionIDOrThrow());
+
+            // If the transaction is not duplicated we want to add it to the cache
+            addTransactionToCache(preHandleResult, recordBuilder.build().recordStreamItem().record(), consensusNow);
```

Comment:
I think we need to add regardless of whether it is a duplicate or not. So you probably need to move the logic in 156 down into the implementation for checkDuplicates, such that we add before we throw in the case of a duplicate, or just add in the case it is not a duplicate.